### PR TITLE
Add rest.getinsomnia.Insomnia

### PR DESCRIPTION
**Pull request by WhyNotHugo**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. https://github.com/Kong/insomnia/issues/1812
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

I'm including a desktop file and an icon file in-tree. I've already mentioned to upstream that it would be idea if these were included in the tarball they ship.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,10 @@
+# Insomnia for Flatpak
```
Comment (barthalion) [inline README.md:1]: No need to have it here unless you feel strongly about it. IMO it just duplicates existing documentation.

File: rest.insomnia.client.yml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  # I'm not a fan of this permission, but for some reason Insomnia won't start
+  # without it. Contributions welcome:
+  - --device=all
+modules:
+  - name: insomnia
+    buildsystem: simple
+    build-options:
+      strip: false
+      no-debuginfo: true
+      no-debuginfo-compression: true
```
Comment (barthalion) [inline rest.insomnia.client.yml:24]:
```suggestion
```

File: rest.insomnia.client.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    build-commands:
+      - tar xf insomnia.tar.gz
+      - install -dm755 /app/share/
+      - install -dm755 /app/main/
+      - mv Insomnia.Core-2021.3.0/* /app/main
```
Comment (barthalion) [inline rest.insomnia.client.yml:29]:
```suggestion
      - mv Insomnia.Core-*/* /app/main
```

File: rest.insomnia.client.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      - install -Dm644 insomnia.desktop /app/share/applications/insomnia.desktop
+      - install -Dm644 insomnia.png /app/share/icons/hicolor/scalable/apps/insomnia.png
+
+      - install -Dm755 insomnia.sh /app/bin/insomnia
+      - install -Dm644 rest.insomnia.client.appdata.xml /app/share/metainfo/$FLATPAK_ID.appdata.xml
+    sources:
+      - type: file
+        dest-filename: insomnia.tar.gz
+        only-arches:
+          - x86_64
```
Comment (barthalion) [inline rest.insomnia.client.yml:40]: Please add https://github.com/flathub/flathub/wiki/App-Maintenance#only-build-on-x86_64 to limit architectures.

File: rest.insomnia.client.appdata.xml [hunk 5]
```diff
@@ -0,0 +1,76 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (barthalion) [inline rest.insomnia.client.appdata.xml:1]: Ideally, this file should be submitted and maintained upstream.

File: insomnia.desktop [hunk 6]
```diff
@@ -0,0 +1,9 @@
+[Desktop Entry]
```
Comment (barthalion) [inline insomnia.desktop:1]: Ideally, this file should be submitted and maintained upstream.

File: rest.insomnia.client.appdata.xml [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    <content_attribute id="language-humor">none</content_attribute>
+    <content_attribute id="language-discrimination">none</content_attribute>
+    <content_attribute id="social-chat">non</content_attribute>
+    <content_attribute id="social-info">none</content_attribute>
+    <content_attribute id="social-audio">non</content_attribute>
+    <content_attribute id="social-location">none</content_attribute>
+    <content_attribute id="social-contacts">intense</content_attribute>
+    <content_attribute id="money-purchasing">none</content_attribute>
+    <content_attribute id="money-gambling">none</content_attribute>
+  </content_rating>
```
Comment (barthalion) [inline rest.insomnia.client.appdata.xml:74]:
```suggestion
  <content_rating type="oars-1.1" />
```

File: rest.insomnia.client.yml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        url: https://github.com/Kong/insomnia/releases/download/core%402021.3.0/Insomnia.Core-2021.3.0.tar.gz
```
Comment (barthalion) [inline rest.insomnia.client.yml:41]: Consider using JSON checker from https://github.com/flathub/flatpak-external-data-checker/ to automate updates.

File: rest.insomnia.client.yml [hunk 9]
```diff
@@ -0,0 +1,64 @@
+app-id: rest.insomnia.client
```
Comment (barthalion) [inline rest.insomnia.client.yml:1]: It probably should be `rest.insomnia.Insomnia`, but it's up to you.

File: rest.insomnia.client.yml [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '20.08'
+rename-desktop-file: insomnia.desktop
+rename-icon: insomnia
+separate-locales: false
+command: insomnia
+finish-args:
+  - --socket=x11
+  - --socket=wayland
```
Comment (barthalion) [inline rest.insomnia.client.yml:13]:
```suggestion
```

I assume it uses xwayland on Wayland.

**Comment by barthalion** [comment]
bot, build rest.insomnia.client

File: README.md (hunk 1 above)
Comment (WhyNotHugo) [reply README.md:1]: Nah, this was just here because this repository as experimental for a long time.

File: insomnia.desktop (hunk 6 above)
Comment (WhyNotHugo) [reply insomnia.desktop:1]: I've requested this [in the upstream discussion](https://github.com/Kong/insomnia/issues/1812#issuecomment-849766586).

It'll remove it from here once upstream updates their tarballs.

File: rest.insomnia.client.yml (hunk 8 above)
Comment (WhyNotHugo) [reply rest.insomnia.client.yml:41]: Oh, that's very handy. Done.

File: rest.insomnia.client.yml (hunk 10 above)
Comment (WhyNotHugo) [reply rest.insomnia.client.yml:13]: Xwayland is a fallback for apps that don't support wayland, and has issues (e.g.: no proper scaling support).

File: rest.insomnia.client.yml (hunk 9 above)
Comment (WhyNotHugo) [reply rest.insomnia.client.yml:1]: Dunny why I went with that name, but your suggestion makes a lot more sense.

File: rest.insomnia.client.appdata.xml (hunk 5 above)
Comment (WhyNotHugo) [reply rest.insomnia.client.appdata.xml]: Isn't this flatpak-specific? I can bring it up upstream, but I'd prefer to know a bit more about what other uses it has.

**Comment by WhyNotHugo** [comment]
I changed the name, so you might have to tell the bot to build under the new name.

File: rest.insomnia.client.yml [hunk 11]
```diff
@@ -0,0 +1,66 @@
...
+  - name: insomnia
+    buildsystem: simple
+    build-commands:
+      - tar xf insomnia.tar.gz
+      - install -dm755 /app/share/
+      - install -dm755 /app/main/
+      - mv Insomnia.Core-*/* /app/main
+
+      - install -Dm644 insomnia.desktop /app/share/applications/insomnia.desktop
+      - install -Dm644 insomnia.png /app/share/icons/hicolor/scalable/apps/insomnia.png
```
Comment (bilelmoussaoui) [inline rest.insomnia.client.yml:28]: why not install them with the right name directly?

File: insomnia.desktop [hunk 12]
```diff
@@ -0,0 +1,9 @@
...
+Name=Insomnia
+Comment=Debug APIs like a human, not a robot
+Exec=/app/bin/insomnia %U
+Terminal=false
+Type=Application
+Icon=insomnia
+MimeType=x-scheme-handler/insomnia;
```
Comment (bilelmoussaoui) [inline insomnia.desktop:8]: Could use some keywords if you want to make the app easily discoverable

**Comment by bilelmoussaoui** [comment]
bot, build rest.insomnia.Insomnia

File: rest.insomnia.client.yml [hunk 13]
```diff
@@ -0,0 +1,66 @@
+app-id: rest.insomnia.Insomnia
```
Comment (bilelmoussaoui) [inline rest.insomnia.client.yml:1]: The app-id != filename. They should be the same.

File: rest.insomnia.client.appdata.xml (hunk 5 above)
Comment (bilelmoussaoui) [reply rest.insomnia.client.appdata.xml]: It's used by any app store on Linux. deb/snap or whatever other packaging format could use it. Even fwupd uses it to describe the firmwares for software centers.

File: insomnia.desktop (hunk 12 above)
Comment (WhyNotHugo) [reply insomnia.desktop:8]: Note sure in which context these are used, shall I just copy the ones upstream uses on their git repo?

```suggestion
MimeType=x-scheme-handler/insomnia;
Keywords=api,graphql,curl,rest-api,grpc,http-client,api-client,electron-app,api-design
```

File: rest.insomnia.client.appdata.xml (hunk 5 above)
Comment (WhyNotHugo) [reply rest.insomnia.client.appdata.xml]: Huh, I'd never seen it on any of the distributions I've worked with.

What's the name of this manifest format? I'll ask upstream if they can ship it in their tarball. They already support `snap`, so they may already have a copy somewhere -- but it's not in the git repo, oddly.

Comment (barthalion) [reply rest.insomnia.client.appdata.xml]: https://www.freedesktop.org/wiki/Distributions/AppStream/

This is not a blocker, just something that would be nice to be handled at some point.

**Comment by barthalion** [comment]
bot, build rest.insomnia.Insomnia

**Comment by barthalion** [comment]
bot, build rest.insomnia.Insomnia

File: rest.insomnia.client.appdata.xml (hunk 5 above)
Comment (WhyNotHugo) [reply rest.insomnia.client.appdata.xml]: Thanks! I'll include this in the upstream PR and mention it adds value on Debian / Snap too. They publish  packages in those formats, so there's good reason to include it.

**Comment by WhyNotHugo** [comment]
Thanks for addressing those final touches!

File: insomnia.desktop (hunk 12 above)
Comment (bilelmoussaoui) [reply insomnia.desktop:8]: they will be used for search on Flathub & other software centers. Also on the gnome shell search menu (and probably other DE as well)

Comment (WhyNotHugo) [reply insomnia.desktop:8]: Sounds like the desktop file and the AppStream have _some_ overlap, right?

**Comment by nedrichards** [comment]
bot, build rest.insomnia.Insomnia

**Comment by barthalion** [comment]
Hi @dimitropoulos, as you were involved in the discussion about flatpak before, do you mind if we give you write access to the Flathub repo? We can also invite other people from your team.

**Comment by WhyNotHugo** [comment]
Build ran around 4 hours ago. Any idea why the package itself isn't showing up as published?

**Comment by hfiguiere** [comment]
it is now.

**Pull request closed without merging**